src/app/components/media/EditTitleAndDescriptionDialog.js

```diff
@@ -0,0 +1,206 @@
+        },
+      },
+    },
+    variables: {
+      input: {
+        id: projectMedia.id,
+        metadata: JSON.stringify(metadata),
+      },
+    },
+    onError: onFailure,
+    onCompleted: ({ data, errors }) => {
+      if (errors) {
+        return onFailure(errors);
+      }
+      return onSuccess(data);
+    },
+  });
+}
+
+/**
+ * Form prompting the user for new title+description.
+ *
+ * On submit, this form calls `onClose()` and starts writing the new values
+ * to the database. If there's an error afterwards, it calls
+ * `setFlashMessage()`.
+ *
+ * Internal state is `null` by default, meaning "the form values will be
+ * whatever is in GraphQL". If the user starts editing, the edits will
+ * persist until the user saves or the form is rendered with `open === false`.
+ */
+function EditTitleAndDescriptionDialog({
+  projectMedia, open, onClose,
+}) {
+  // State is null by default: if the data changes, the form will update.
+  const [newTitle, setNewTitle] = React.useState(null);
+  const [newDescription, setNewDescription] = React.useState(null);
+  const setFlashMessage = React.useContext(FlashMessageSetterContext);
+  const handleChangeTitle = React.useCallback(
+    ev => setNewTitle(ev.target.value),
+    [setNewTitle],
+  );
+  const handleChangeDescription = React.useCallback(
+    ev => setNewDescription(ev.target.value),
+    [setNewDescription],
+  );
+
+  const formId = `edit-title-and-description-${projectMedia.dbid}`;
+
+  React.useEffect(() => {
+    // If ever the caller closes the dialog, clear its state.
+    if (!open) {
+      setNewTitle(null);
+      setNewDescription(null);
+    }
+    // If ever a user update or server update makes the user-visible value
+    // the same as the api-side value, clear its state.
+    if (newTitle === projectMedia.metadata.title) {
+      setNewTitle(null);
+    }
+    if (newDescription === projectMedia.metadata.description) {
+      setNewDescription(null);
+    }
+  }, [open, newTitle, newDescription, setNewTitle, setNewDescription]);
+
+  const title = newTitle === null ? projectMedia.metadata.title : newTitle;
+  const description = newDescription === null ? projectMedia.metadata.description : newDescription;
+
+  const handleSubmit = React.useCallback((ev) => {
+    ev.preventDefault();
+    onClose(); // useEffect() will clear state
+    commitSetProjectMediaTitleAndDescription({
+      projectMedia,
+      title,
+      description,
+      onSuccess: () => {},
+      onFailure: (errors) => {
+        console.error(errors); // eslint-disable-line no-console
+        setFlashMessage((
+          <FormattedMessage
+            id="mediaDetail.editReportError"
+            defaultMessage="Sorry, an error occurred while updating the item. Please try again and contact {supportEmail} if the condition persists."
+            values={{ supportEmail: stringHelper('SUPPORT_EMAIL') }}
+          />
+        ));
+      },
```

Comment:
Our typical handling is to extract a message from the request response and use unknownError or a more context-specific message as a fallback.